scripts/irods/test/test_iadmin.py

```diff
@@ -2535,3 +2535,159 @@ def test_mkgroup_with_remote_zone(self):
         finally:
             self.admin.run_icommand(['iadmin', 'rmgroup', self.group])
             self.admin.run_icommand(['iadmin', 'rmzone', remote_zone])
+
+
+class test_mkzone_conn_str_validation(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        """Set up the test class."""
+        cls.admin = session.mkuser_and_return_session('rodsadmin', 'otherrods', 'rods', lib.get_hostname())
+
+    @classmethod
+    def tearDownClass(cls):
+        """Tear down the test class."""
+        with session.make_session_for_existing_admin() as admin_session:
+            cls.admin.__exit__()
+            admin_session.assert_icommand(['iadmin', 'rmuser', cls.admin.username])
+
+    def test_good_localhost(self):
+        try:
+            self.admin.assert_icommand(['iadmin', 'mkzone', 'goodzone', 'remote', 'localhost:1247'], 'EMPTY')
```

Comment:
Should we confirm the existence (or non-existence) of the zones as a result of the success (or failure) of `mkzone` for these tests?

Reply:
Good idea